Sources/drawing.py

```diff
@@ -0,0 +1,418 @@
+        observations = [entity for entity in entities if entity.EntityType == 2]
+        nodeInspections = [entity for entity in entities if entity.EntityType == 5]
+        nodeObservations = [entity for entity in entities if entity.EntityType == 6]
+
+        self.delete_features(sections, self.created_layers[1])
+        self.delete_features(nodes, self.created_layers[2])
+        self.delete_features(inspections, self.created_layers[0])
+        self.delete_features(observations, self.created_layers[5])
+        self.delete_features(nodeInspections, self.created_layers[3])
+        self.delete_features(nodeObservations, self.created_layers[4])
+        self.qgis.mapCanvas().refresh()
+                                       
+    def get_selected(self):
+        count = 0
+        selected_shapes = []
+        layer = self.qgis.activeLayer()
+        if type(layer) is not type(None):
+            for feature in layer.selectedFeatures():
+                count += 1
+                selected_shapes.append(feature)
+        return count, selected_shapes, layer
+    
+    def draw_Inspections(self, Inspections):
+        QCoreApplication.processEvents()
+        inspections = []
+        InspectionLayer = self.created_layers[0]
+        InspectionLayer.startEditing()
+        prov = InspectionLayer.dataProvider()
+        for ins in Inspections:
+            points = []
+            feature = QgsFeature()
+            feature.setFields(self.fields[0])
+            for point in ins.Vertices:
+                points.append(QgsPointXY(point.X, point.Y))
+            feature.setGeometry(QgsGeometry.fromPolylineXY(points))
+            self.add_values(
+                feature, ins, self.VX.LayerFieldsPovider.InspectionShapeFields)
+            inspections.append(feature)
+
+        prov.addFeatures(inspections)
+        InspectionLayer.updateExtents()
+        InspectionLayer.commitChanges()
+        QCoreApplication.processEvents()
+
+    def draw_Sections(self, Sections):
+        QCoreApplication.processEvents()
+        nr = 1
+        sections = []
+        SectionLayer = self.created_layers[1]
+        SectionLayer.startEditing()
+        prov = SectionLayer.dataProvider()
+        points = []
+
+        if SectionLayer == None:
+            return
+
+        for section in Sections:
+
+            if section.ExternalItemId != 0:
+                request = QgsFeatureRequest().setSubsetOfAttributes(
+                    ["OBJ_PK"], SectionLayer.fields()).setFilterExpression('"OBJ_PK"=\'%s\'' % section.Id)
+                features = SectionLayer.getFeatures(request)
+                for f in features:
+                    for value in self.VX.LayerFieldsPovider.SectionShapeFields:
+                        attr = SectionLayer.fields().indexFromName(str(value.Key))
+                        SectionLayer.changeAttributeValue(
+                            f.id(), attr, str(section.GetValue(value)))
```

Comment:
for/if/for/for I am already lost. Can we make it a function to give a chance to understand what this is doing more expressively?

Reply:
Changed into method